OpenSim/Common/Component.h

```diff
@@ -485,6 +485,15 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
         return *const_cast<C*>(&(this->template getComponent<C>(name)));
     }
 
+    /** Print a list to the console of all components whose full path name
+     * contains the given string. You might use this if (a) you know the name of
+     * a component in your model but don't know its full path, (b) if you want
+     * to find all components with a given name, or (c) to get a list of all
+     * components on the right leg of a model (if all components on the right
+     * side have "_r" in their name).
+     * @returns The number of matches. */
+    int printComponentsMatching(const std::string& substring);
```

Comment:
Good points with (1) and (3). 

My intent with (2) is that advanced users can use ComponentFilterFullPathNameRegex directly, but that this method is a convenience when you just want to check for a substring (and is also available in python/java/matlab--std::regex does not get wrapped). I could write a new ComponentFilterFullPathNameContainsSubstring or add another method on Component that takes a regex. What do you think?